src/scenes/Partnerships/Edit/EditPartnership.tsx

```diff
@@ -0,0 +1,117 @@
+import { Box, Button } from '@material-ui/core';
+import React, { FC } from 'react';
+import { Except } from 'type-fest';
+import {
+  displayPartnershipStatus,
+  GQLOperations,
+  PartnershipStatuses,
+  PartnershipType,
+  UpdatePartnershipInput,
+} from '../../../api';
+import {
+  DialogForm,
+  DialogFormProps,
+} from '../../../components/Dialog/DialogForm';
+import {
+  CheckboxesField,
+  CheckboxOption,
+  RadioField,
+  RadioOption,
+  SubmitError,
+} from '../../../components/form';
+import {
+  EditPartnershipFragment,
+  useDeletePartnershipMutation,
+  useUpdatePartnershipMutation,
+} from './EditPartnership.generated';
+
+type EditPartnershipProps = Except<
+  DialogFormProps<UpdatePartnershipInput>,
+  'onSubmit' | 'initialValues'
+> & {
+  partnership: EditPartnershipFragment;
+};
+
+export const EditPartnership: FC<EditPartnershipProps> = ({
+  partnership,
+  ...props
+}) => {
+  const [updatePartnership] = useUpdatePartnershipMutation();
+  const [deletePartnership] = useDeletePartnershipMutation();
+
+  const onDeleteClick = () => {
+    const { id } = partnership;
+    deletePartnership({
+      variables: { input: id },
+      refetchQueries: [GQLOperations.Query.ProjectPartnerships],
+    });
+    //TODO: correctly handle closing the dialog.  props.onClose has a TS errors
+    // props.onClose && props.onClose('cancel');
+  };
+
+  const radioOptions = PartnershipStatuses.map((status) => (
+    <RadioOption label={displayPartnershipStatus(status)} value={status} />
+  ));
+
+  return (
+    <DialogForm<UpdatePartnershipInput>
+      //TODO: fix the form flashing back to initialValues
+      onSubmit={(input) => {
+        updatePartnership({
+          variables: { input },
+          refetchQueries: [GQLOperations.Query.ProjectPartnerships],
+        });
+      }}
+      {...props}
```

Comment:
Ok just FYI here. The order in which props are spread and what comes before and after determines the default values and overrides. The logic that this order defines should match the TS interface. 
This component's props exclude `onSubmit` and `initialValues` from our `DialogFormProps`, because they are handled here. So those should come after the spreading of `props`. Even if you pass them in they should do nothing because we enforce them here.
The opposite is true of passing in `DialogProps`. Those are just defaults, so we don't exclude them from our props interface and they should come before the props spread so if/when they are passed in they are correctly applied.

Does that make sense?

Reply:
Yes it does, thanks for the explanation.  Changed.